google-cloud-storage/src/main/java/com/google/cloud/storage/OpenTelemetryBootstrappingUtils.java

```diff
@@ -88,15 +89,41 @@ final class OpenTelemetryBootstrappingUtils {
 
   static final Logger log = Logger.getLogger(OpenTelemetryBootstrappingUtils.class.getName());
 
-  static void enableGrpcMetrics(
-      InstantiatingGrpcChannelProvider.Builder channelProviderBuilder,
+  @NonNull
+  static ChannelConfigurator enableGrpcMetrics(
+      @Nullable ChannelConfigurator channelConfigurator,
       String endpoint,
-      String projectId,
+      @Nullable String projectId,
       String universeDomain,
       boolean shouldSuppressExceptions) {
+    GCPResourceProvider resourceProvider = new GCPResourceProvider();
+    Attributes detectedAttributes = resourceProvider.getAttributes();
+
+    @Nullable String detectedProjectId =
+        detectedAttributes.get(AttributeKey.stringKey("cloud.account.id"));
+    if (projectId == null && detectedProjectId == null) {
+      log.warning(
+          "Unable to determine the Project ID in order to report metrics. No gRPC client metrics will be reported.");
+      return channelConfigurator != null ? channelConfigurator : ChannelConfigurator.identity();
```

Comment:
qq: is this the case of not running on GCP?

Reply:
In a round about manner yes. The metrics reporting relies on application default credentials for initialization, and if the project can't be determined either as a passed in value from the Storage client, or resolved from the GCP resolved otel attributes there isn't any way to bootstrap.